### PR TITLE
Add support for `freq=None` in missing_* functions

### LINKED ISSUE #432: Support `freq=None` in missing_* functions
### Description

This will simplify the Indicator class logic and facilitate switching the missing method.

### PR DESCRIPTION
**Pull request by huard**
### Pull Request Checklist:
- [x] This PR addresses an already opened issue (for bug fixes / features)
    - This PR fixes #432
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Documentation has been added / updated (for bug fixes / features)
- [ ] HISTORY.rst has been updated (with summary of main changes)
- [ ] `bumpversion (minor / major / patch)` has been called on this branch
- [ ] Tags have been pushed (`git push --tags`)

* **What kind of change does this PR introduce?** 
missing functions can now handle freq=None, which is useful when computing climatologies. 

* **Does this PR introduce a breaking change?** 
No

* **Other information**:

### CODE REVIEW

File: xclim/indicators/atmos/_conversion.py [hunk 1]
```diff
@@ -19,6 +19,10 @@ class Converter(Indicator):
     def validate(self, da):
         """Input validation."""
 
+    def missing(self, *args, **kwds):
+        """Return whether an output is considered missing or not."""
+        return False
```
Comment (aulemahal) [inline xclim/indicators/atmos/_conversion.py:24]: Not sure why we would skip checks on "_conversion" indicators? Usually we can trust xarray to do it anyway, but what if?  An invalid "tasmin" should imply an invalid "tas", and so on.

Comment (huard) [reply xclim/indicators/atmos/_conversion.py:24]: Totally agree, but `missing` is intended for data reductions along time. In this case, we have a scalar operation (n -> n), so I assume there is no need for the missing function. This is a check I could add.

Comment (aulemahal) [reply xclim/indicators/atmos/_conversion.py:30]: Euh, but isn't that what this PR adds? Missing methods for non-reducing indicators?

Comment (huard) [reply xclim/indicators/atmos/_conversion.py:24]: No, it's missing methods for indicators reducing on entire series (no resampling): n -> 1

**Pull request merged**